nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentPropertiesJsonReader.java

```diff
@@ -187,19 +191,20 @@ private Object getScalarPropertyValue(Property property, JsonNode jn, Type type)
             ObjectResolver resolver = type.getObjectResolver();
             if (resolver == null) {
                 // Let's assume it is a blob of which content has to be stored in a string property.
-                assert type.getSuperType() instanceof StringType;
-                Blob blob = readEntity(Blob.class, Blob.class, jn);
-                if (blob == null) {
-                    throw new MarshallingException("Unable to parse the property " + property.getXPath());
+                if (type.getSuperType() instanceof StringType) {
+                    Blob blob = readEntity(Blob.class, Blob.class, jn);
+                    if (blob != null) {
+                        return blob.getString();
+                    }
                 }
-                return blob.getString();
+                log.info("Unable to parse the property {}", property::getXPath);
```

Comment:
warn maybe?

Reply:
I asked myself the same question..

The thing is `warn` will be suddenly displayed after this fix for use case like the unit test one and we'll end up with a warning for an expected use case

Reply:
If this use case  only happens in tests you can hide the warning with the `LogFeature`, using `ConsoleLogLevelThreshold` on master and `hideWarningFromConsoleLog`/`restoreConsoleLog` before that.

Reply:
It does not only happen in tests but in real use case too (the unit tests reproduces the steps described in https://jira.nuxeo.com/browse/NXP-27932?focusedCommentId=411451&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-411451)

Reply:
OK so info :+1: